src/AstrometryFit.cc

```diff
@@ -549,66 +549,96 @@ void AstrometryFit::checkStuff() {
-          << "residual in degrees in TP" << separator << separator << "transformed coordinate in TP"
-          << separator << separator << "rough mag" << separator << "Modified Julian date of the measurement"
-          << separator << "transformed measurement uncertainty" << separator << separator << separator
-          << "currently unused" << separator << "unique index of the fittedStar" << separator
-          << "on sky position of fittedStar" << separator << separator << "contribution to Chi2 (2D dofs)"
-          << separator << "number of measurements of this fittedStar" << separator << "chip id" << separator
-          << "visit id" << std::endl;
+
+    ofile << "#id" << separator << "xccd" << separator << "yccd " << separator;
+    ofile << "rx" << separator << "ry" << separator;
+    ofile << "xtp" << separator << "ytp" << separator;
+    ofile << "mag" << separator << "mjd" << separator;
+    ofile << "xErr" << separator << "yErr" << separator << "xyCov" << separator;
+    ofile << "xtpi" << separator << "ytpi" << separator;
+    ofile << "rxi" << separator << "ryi" << separator;
+    ofile << "color" << separator << "fsindex" << separator;
+    ofile << "ra" << separator << "dec" << separator;
+    ofile << "chi2" << separator << "nm" << separator;
+    ofile << "chip" << separator << "visit" << std::endl;
+
+    ofile << "#id in source catalog" << separator << "coordinates in CCD (pixels)" << separator << separator;
+    ofile << "residual on TP (degrees)" << separator << separator;
+    ofile << "transformed coordinate in TP (degrees)" << separator << separator;
+    ofile << "rough magnitude" << separator << "Modified Julian Date of the measurement" << separator;
+    ofile << "transformed measurement uncertainty (degrees)" << separator << separator << separator;
+    ofile << "as-read position on TP (degrees)" << separator << separator;
+    ofile << "as-read residual on TP (degrees)" << separator << separator;
+    ofile << "currently unused" << separator << "unique index of the fittedStar" << separator;
+    ofile << "on sky position of fittedStar" << separator << separator;
+    ofile << "contribution to Chi2 (2D dofs)" << separator << "number of measurements of this fittedStar"
+          << separator;
+    ofile << "chip id" << separator << "visit id" << std::endl;
+
     const CcdImageList &ccdImageList = _associations->getCcdImageList();
     for (auto const &ccdImage : ccdImageList) {
         const MeasuredStarList &cat = ccdImage->getCatalogForFit();
         const Mapping *mapping = _astrometryModel->getMapping(*ccdImage);
+        const auto readTransfo = ccdImage->readWCS();
         const Point &refractionVector = ccdImage->getRefractionVector();
         double mjd = ccdImage->getMjd() - _JDRef;
         for (auto const &ms : cat) {
             if (!ms->isValid()) continue;
             FatPoint tpPos;
             FatPoint inPos = *ms;
+            FatPoint inputTpPos;
```

Comment:
Consider moving this to just before it's initialized by `apply`. It will make the code easier to understand (I had to use text search to find where the variable was used), and possibly more efficient.

(It would also be better if you could initialize `inputTpPos` at declaration time, but I assume the API of `gtransfoCompose` prevents you from doing that.)

Reply:
Turns out, I could do it exactly like your parenthetical, so I did!